stellar-contract-env-host/src/host.rs

```diff
@@ -393,6 +397,42 @@ impl Host {
             key: sckey,
         })
     }
+
+    #[cfg(feature = "vm")]
+    fn call_n(&self, contract: Object, func: RawVal, args: &[ScVal]) -> Result<RawVal, HostError> {
+        // Create key for storage
+        let id = self.visit_obj(contract, |bin: &Vec<u8>| {
+            let arr: [u8; 32] = bin
+                .as_slice()
+                .try_into()
+                .map_err(|_| HostError::General("invalid contract hash"))?;
+            Ok(ContractID(xdr::Hash(arr)))
+        })?;
+        let key = ScVal::Static(ScStatic::Void); // TODO: replace with "SCS_LEDGER_KEY_CONTRACT_CODE_WASM"
+        let storage_key = Key {
+            contract_id: id.clone(),
+            key,
+        };
+        // Retrieve the contract code and create vm
+        let scval = self.0.storage.borrow_mut().get(&storage_key)?;
+        let scobj = match scval {
+            ScVal::Object(Some(ob)) => *ob,
+            _ => return Err(HostError::General("not an object")),
+        };
+        let code = match scobj {
+            ScObject::Binary(b) => b,
+            _ => return Err(HostError::General("not a binary object")),
+        };
+        let vm = Vm::new(&self, id, code.as_slice())?;
+        // Resolve the function symbol and invoke contract call
+        let sym = unsafe { <Symbol as RawValConvertible>::unchecked_from_val(func) };
```

Comment:
You can actually set the argument type for `func` to be `Symbol`, and then it will automatically cause a conversion error if a user calls one of the `call_{1,2,3,4}` functions with a non-symbol value, and you can get rid of this `unchecked_from_val`.

Reply:
That's great. Fixed!